airbyte-integrations/connectors/source-shopify/source_shopify/spec.json

```diff
@@ -11,6 +11,7 @@
         "type": "string",
         "title": "Shopify Store",
         "description": "The name of your Shopify store found in the URL. For example, if your URL was https://NAME.myshopify.com, then the name would be 'NAME'.",
+        "pattern": "^(?!https:\/\/)(?!http:\/\/)(?!.*\\.myshopify\\.)(?!.*\\.myshopify).*",
```

Comment:
The current regex (validated with https://rubular.com/) allows `company-name.shopify.com`, which I don't think is desired (see https://rubular.com/r/pG2m72BxZLCxFb), and I don't think the "capture" statements are working.  If the goal is only to allow NAME (per the description), perhaps the pattern should also ignore any `.` characters as well?

Reply:
The regex now will validate whether or not the user inputs some URLs which we cannot actually process correctly,  since there is no problem with having the `shop name` = `my.shop.name.myshopify.com`, where `my.shop.name` as the name we need. I'm also trying to check the Shopify restrictions for the `shop names` (specifically, what symbols are allowed), so we are aligned with their rules to match the shop name pattern effectively. No doubts  - this solution is step one, I'm sure we will improve this in the future as well as will hit some edge cases, as always.

Reply:
Thus, if the input string contains `https' or `myshopify` (+ variations with prefix and suffix dots) we are good to catch that user inputs something we cannot actually handle correctly, so extracting the name of the shop should be done in prior and the field should only accept the effective name, not URLs. or similar like not  (https://my_store.myshopify.com/ or  my_store.myshopify.com) 
there are tons of variations that could be made here, in order to minimize this variation, we need to strict this field to something that could be easily processed from our side.